### PR TITLE
feat: migrate test to playwright and add github action

### PR DESCRIPTION
**Pull request by TrickyPi**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        with:
+          node-version: "14"
+      - name: Install dependencies
+        run: npm ci
+      - name: Install Playwright
+        run: npx playwright install --with-deps
+      - name: build
+        run: npm run build
+      - name: Run your tests
+        run: npm test
```
Comment (jpillora) [inline .github/workflows/ci.yml:24]: is it worth testing on windows (edge) as well? should be fairly easy with a test matrix https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#using-a-matrix-strategy

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,4 @@
 node_modules
 bower_components
+
```
Comment (jpillora) [inline .gitignore:3]: can we also add `.DS_Store` (and delete the file in this PR)

File: tests/xhr-web-worker.spec.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+//   await t
+//     .switchToIframe('iframe')
+//     .expect(Selector('#one').innerText).eql('this is the first text file example (example1.txt)');
+// });
+
+// test('example 2 response should be modified', async t => {
+//   await t
+//     .switchToIframe('iframe')
+//     .expect(Selector('#two').innerText).eql('thzs zs thz szcznd tzxt fzlz zxzmplz (zxzmplz2.txt)');
+// });
```
Comment (jpillora) [inline tests/xhr-web-worker.spec.ts:16]: im happy to delete the old test, we can always get it back from git history

**Review by jpillora — CHANGES_REQUESTED** [review]
overall, its looking really good 🚀

File: .gitignore (hunk 2 above)
Comment (TrickyPi) [reply .gitignore:3]: done

File: tests/xhr-web-worker.spec.ts (hunk 3 above)
Comment (TrickyPi) [reply tests/xhr-web-worker.spec.ts]: done

File: .github/workflows/ci.yml (hunk 1 above)
Comment (TrickyPi) [reply .github/workflows/ci.yml:27]: done

**Comment by TrickyPi** [comment]
Currently, some tests fail randomly, e.g. the `real-download-events` test, but I think we can merge this pr first and then fix these tests.

**Comment by jpillora** [comment]
With no secrets in the repo, I think we can enable tests on all PRs

**Comment by TrickyPi** [comment]
Yeah.

**Pull request merged**